daemon/mgr/container_envs.go

```diff
@@ -0,0 +1,181 @@
+// In original ways, processes in container can only accepted env when startup.
+// And container engine would store these data in container.json which is persistent on local disk.
+// There is no env ralated data in container or image.
+//
+// While for rich container mode, env management in container is a little bit complicated.
+// First, introduce some scenarios in practice:
+// * init process in rich container will not pass envs to user-specific CMD process if user switched from root;
+// * newly created shell command should herit all envs of the container, so all envs should be stored in dir
+// /etc/profild.d/*.sh (here is pouchenv.sh). After that a process execed from outside would take env of init process,
+// then such action like reload of detailed user application could share the env.
+const (
+	// containerInstanceInfo is file in the container image.
+	// row in this file has content like `env_appDeployType = taoJavaWeb`
+	containerInstanceInfo = "/etc/instanceInfo"
+	// pouchEnvDir =
+	pouchEnvDir = "/etc/profile.d"
+	// this filepath is used in PouchContainer to store user input env via persistent file
+	pouchEnvFile = "/etc/profile.d/pouchenv.sh"
+)
+
+// updateContainerEnv update the container's envs in
+// /etc/instanceInfo and /etc/profile.d/pouchenv.sh
+// Env used by rich container.
+func updateContainerEnv(inputRawEnv []string, baseFs string) error {
+	var (
+		envPropertiesPath = path.Join(baseFs, containerInstanceInfo)
+		envShPath         = path.Join(baseFs, pouchEnvFile)
+	)
+
+	// check the existence of related files.
+	// if dir of pouch.sh is not exist, it's unnecessary to update that files.
+	if _, err := os.Stat(path.Join(baseFs, pouchEnvDir)); err != nil {
+		return nil
+	}
+	if _, err := os.Stat(envPropertiesPath); err != nil {
+		//if etc/instanceInfo is not exist, it's unnecessary to update that file.
+		return nil
+	}
+	if _, err := os.Stat(envShPath); err != nil {
+		return fmt.Errorf("failed to stat container's env file /etc/profile.d/pouchenv.sh: %v", err)
+	}
+
+	inputEnv := utils.ConvertKVStrToMapWithNoErr(inputRawEnv)
+
+	localEnv, err := getLocalEnv(envShPath)
+	if err != nil {
+		return fmt.Errorf("failed to get local env from file %s: %v", envShPath, err)
+	}
+
+	newEnv := combineLocalAndInputEnv(inputEnv, localEnv)
+
+	// start to construct new content for file pouchenv.sh after combining
+	// original local envs extracted from pouchenv.sh and input envs.
+	var str string
+	for key, value := range newEnv {
+		str += fmt.Sprintf("export %s=\"%s\"\n", key, value)
+	}
+	ioutil.WriteFile(envShPath, []byte(str), 0755)
+
+	p, err := properties.LoadFile(envPropertiesPath, properties.ISO_8859_1)
+	if err != nil {
+		return fmt.Errorf("failed to properties load container's environment variable file(/etc/instanceInfo): %v", err)
+	}
+
+	for key, val := range newEnv {
+		if v, ok := p.Get("env_" + key); ok {
+			if key == "PATH" {
+				// refer to https://aone.alipay.com/project/532482/task/9745028
```

Comment:
it's my fault to add this url to PouchContainer's code, can you help me delete this line, thanks ? @allencloud